### PR TITLE
Closes #164 - Time-constrained top frecent links API

### LINKED ISSUE #164: Time-constrained top frecent links API
We need a new API to power the "top activity" rubric instead of the recent bookmark API.

Here's a cartoon sketch of a query, using postgres operators for datetime. Probably needs work.

```sql
SELECT p.url, p.title, p.frecency,
               p.last_visit_date as lastVisitDate,
                b.id as bookmarkId,
                b.guid as bookmarkGuid,
                f.data as favicon,
                f.mime_type as mimeType,
                "history" as type
FROM moz_places p
LEFT JOIN moz_bookmark m
ON m.fk = p.id
LEFT JOIN moz_favicons f
ON p.favicon_id = f.id
WHERE p.last_visit_date > current_date - interval '72' hour
AND  p.hidden = 0 AND p.last_visit_date NOTNULL
ORDER BY frecency DESC, lastVisitDate DESC, url;
```

It is a history type of query, with bookmarkGuid and id as a marker of whether or not the link has been bookmarked.

The context surrounding this task is explained in #158.

### PR DESCRIPTION
**Pull request by mzhilyaev**
Note a change to timeDaysAgo() - it will return microseconds instead of milliseconds.
This is done because FX keeps last_visit_date in microseconds

Fixes #164

### CODE REVIEW

File: lib/PlacesProvider.js [hunk 1]
```diff
@@ -223,6 +245,32 @@ Links.prototype = {
   }),
 
   /**
+   * Gets the top recent links
+   *
+   * @param {Integer} options
+   *                  limit: Maximum number of results to return. Max 100.
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: Max 25 by default, right?

File: lib/PlacesProvider.js [hunk 2]
```diff
@@ -180,20 +183,38 @@ Links.prototype = {
   },
 
   /**
-   * Gets the top recent links
+   * Gets links from history based on supplied options
    *
-   * @param {Integer} options
-   *                  limit: Maximum number of results to return. Max 100.
+   * @param {Object} options
+   *        {Integer} limit: Maximum number of results to return. Max 100.
```
Comment (oyiptong) [inline lib/PlacesProvider.js]: Max seems to be 25 now

**Comment by oyiptong** [comment]
r+ with nits

File: lib/PlacesProvider.js (hunk 1 above)
Comment (k88hudson) [inline lib/PlacesProvider.js]: Yeah, this is temporary til we get post-style requests with embedly

**Comment by mzhilyaev** [comment]
Closing this PR as it needs to be rebased on top of master

**Pull request closed without merging**